src/utils/OneSignalUtils.ts

```diff
@@ -110,6 +111,17 @@ export class OneSignalUtils {
   public static logMethodCall(methodName: string, ...args: any[]) {
     return Log.debug(`Called %c${methodName}(${args.map(Utils.stringify).join(', ')})`, Utils.getConsoleStyle('code'), '.');
   }
+
+  static isHttps(): boolean {
+    if (OneSignalUtils.isSafari()) {
+      return window.location.protocol === "https:";
+    }
+    return !OneSignalUtils.isUsingSubscriptionWorkaround();
```

Comment:
was surprised this is part of the 'isHttps' check -- is there any chance of https with the subscription workaround (that isn't safari)?

Reply:
we now have new class `EnvironmentInfo` which contains correct set of things. https for actual https, etc. I'll refactor the code to use it instead of in-place checks.